src/components/Tables/Table.cy.tsx

```diff
@@ -0,0 +1,235 @@
+        cy.get("input[aria-label='Select row 0']").click();
+        cy.get("input[placeholder='Filter by Name']").type("Sam");
+        cy.get("input[placeholder='Filter by Name']").clear();
+        cy.get("input[aria-label='Select row 0']").should("be.checked");
+    });
+
+    it("changing page does not affect checkbox", () => {
+        cy.mount(<Table data={data} headers={headers} />);
+        cy.get("input[aria-label='Select row 0']").should("not.be.checked");
+        cy.get("input[aria-label='Select row 0']").click();
+        cy.get("button[id='pagination-next-page']").click();
+        cy.get("button[id='pagination-previous-page']").click();
+        cy.get("input[aria-label='Select row 0']").should("be.checked");
+    });
+
+    it("pagination limit does not affect checkbox", () => {
+        cy.mount(<Table data={data} headers={headers} />);
+        cy.get("select[aria-label='Rows per page:']").select("15");
+        cy.get("input[aria-label='Select row 14']").should("not.be.checked");
+
+        cy.get("input[aria-label='Select row 14']").click();
+        cy.get("input[aria-label='Select row 14']").should("be.checked");
+
+        cy.get("input[aria-label='Select row 0']").click();
+        cy.get("input[aria-label='Select row 0']").should("be.checked");
+
+        cy.get("select[aria-label='Rows per page:']").select("10");
+        cy.get("input[aria-label='Select row 0']").should("be.checked");
+        cy.get("select[aria-label='Rows per page:']").select("15");
+
+        cy.get("input[aria-label='Select row 0']").should("be.checked");
+        cy.get("input[aria-label='Select row 14']").should("be.checked");
+    });
+
+    it("sorting does not affect checkbox", () => {
+        cy.mount(<Table data={data} headers={headers} />);
+
+        cy.get("input[aria-label='Select row 0']").click();
+        cy.get("input[aria-label='Select row 0']").should("be.checked");
+
+        cy.get("input[aria-label='Select row 2']").click();
+        cy.get("input[aria-label='Select row 2']").should("be.checked");
+
+        cy.get("div").contains("Name").parent().click();
+        cy.get("input[aria-label='Select row 2']").should("be.checked");
+        cy.get("div").contains("Name").parent().click();
+
+        cy.get("input[aria-label='Select row 0']").should("be.checked");
+        cy.get("input[aria-label='Select row 2']").should("be.checked");
+    });
+
+    it("checkall box toggles all data", () => {
+        cy.mount(<Table data={smallerData} headers={headers} />);
+        cy.get("input[aria-label='Select all rows']").click();
+        cy.get("input[aria-label='Select row 0']").should("be.checked");
+        cy.get("input[aria-label='Select row 1']").should("be.checked");
+        cy.get("input[aria-label='Select row 2']").should("be.checked");
+    });
+
+    it("uncheck one row unticks the checkall box", () => {
+        cy.mount(<Table data={smallerData} headers={headers} />);
```

Comment:
do you actually need `smallerData` here? why does `data` itself not work?

Reply:
It makes the toggle check box tests less lengthy